x-pack/plugins/uptime/server/lib/adapters/monitors/elasticsearch_monitors_adapter.ts

```diff
@@ -187,35 +180,42 @@ export class ElasticsearchMonitorsAdapter implements UMMonitorsAdapter {
       },
     };
 
-    const queryResult = await this.database.search(request, params);
-    const hostBuckets = get(queryResult, 'aggregations.urls.buckets', []);
-    const monitorStatuses = hostBuckets.map(bucket => {
-      const latest = get(bucket, 'latest.hits.hits', []);
-      return latest.reduce(
-        (acc, doc) => {
-          const status = get(doc, '_source.monitor.status', null);
-          if (statusFilter && statusFilter !== status) {
-            return acc;
-          }
-          if (status === 'up') {
-            acc.up += 1;
-          } else {
-            acc.down += 1;
-          }
-          return acc;
-        },
-        { up: 0, down: 0 }
-      );
-    });
-    const { up, down } = monitorStatuses.reduce(
-      (acc, status) => {
-        acc.up += status.up;
-        acc.down += status.down;
-        return acc;
-      },
-      // @ts-ignore TODO update typings and remove this comment
-      { up: 0, down: 0 }
-    );
+    let up: number = 0;
+    let down: number = 0;
+    let searchAfter: any = null;
+
+    do {
+      if (searchAfter) {
+        set(params, 'body.aggs.ids.composite.after', searchAfter);
+      }
+
+      const queryResult = await this.database.search(request, params);
+      const idBuckets = get(queryResult, 'aggregations.ids.buckets', []);
+
+      if (idBuckets.length === 0) {
+        break;
+      }
+
+      idBuckets.forEach(bucket => {
+        const latest = get(bucket, 'latest.hits.hits', []);
+        return latest.forEach(
```

Comment:
We're getting close here 😃 

This `forEach` should be removed IMO.

`latest.hits.hits` will always have [exactly 1 item](https://github.com/elastic/kibana/blob/a1a255e671f9fce0eed4092ac3bb61b1db3d7bfa/x-pack/plugins/uptime/server/lib/adapters/monitors/elasticsearch_monitors_adapter.ts#L174), so we should simply `get(bucket, 'latest.hits.hits[0]')`, or even better `get(bucket, 'latest.hits.hits[0]._source.monitor.status')` and do our comparison/incrementing right here. We'll end up with nicer and simpler code.

Here's an implementation that's passing tests for me:
```
idBuckets.forEach(bucket => {
  const status = get(bucket, 'latest.hits.hits[0]._source.monitor.status');
  if (!statusFilter || (statusFilter && statusFilter === status)) {
    if (status === 'up') {
      up++;
    } else {
      down++;
    }
  }
});
```

Note I added an extra clause to the first `if` condition.. without that we bust if no status filter is present.

Reply:
Great points. All cleaned up in 17b2f8f